### PR TITLE
Re-Support old hotkey mode behaviour

### PR DESCRIPTION
**Pull request by ninjanomnom**
:cl: ninjanomnom
fix: You can use the old non-hotkey mode text input behavior (keyboard input automatically directed at the text input box) by changing your hotkey mode in preferences
/:cl:

The implementation is bad to look at but it works. It can be done better when someone feels like datumizing all this and setting up custom controls.

This also moves the macros list to SSinput and improves the method used to remove all existing macros from a user.

closes #33310

### CODE REVIEW

File: code/controllers/subsystem/input.dm [hunk 1]
```diff
@@ -1,10 +1,103 @@
 SUBSYSTEM_DEF(input)
 	name = "Input"
 	wait = 1 //SS_TICKER means this runs every tick
-	flags = SS_TICKER | SS_NO_INIT | SS_KEEP_TIMING
+	init_order = INIT_ORDER_INPUT
+	flags = SS_TICKER | SS_KEEP_TIMING
 	priority = 151
 	runlevels = RUNLEVELS_DEFAULT | RUNLEVEL_LOBBY
 
+	var/initialized = FALSE
```
Comment (ninjanomnom) [inline code/controllers/subsystem/input.dm]: this'll be removed when the other pr gets through

**Comment by nicbn** [comment]
can you explain better what is old hotkey mode behaviour?

**Comment by ninjanomnom** [comment]
The ability to type while moving. I didn't want to do this as it forces us to have multiple macro sets until full custom controls are implemented.

**Comment by Nabski89** [comment]
When I have hotkey mode on my options for movement seem to be
Arrow keys (ha no)
Have active chat typing going, then control wasd.

If there isn't active typing going on I just spin in retarded circles. Is this the intended functionality?

**Comment by ninjanomnom** [comment]
You'll need to explain better what's happening

**Comment by Nabski89** [comment]
image
Doesn't let you move with control wasd
image
Lets you move with control wasd

**Comment by ninjanomnom** [comment]
I still don't get what you're saying is the issue or what you're trying to accomplish.

The game preferences option is what controls the mode switch between the old hotkey mode behavior and the new. Tab acts differently depending on which you're using. Hotkey mode in preferences means all tab does is focus/unfocus text input. Old behavior makes it actually change the hotkeys. 

In the new mode ctrl+wasd will always make you turn.

**Comment by Nabski89** [comment]
Do you see how there's the little typing menu in the second picture?

When that's there the old behavior works fine. Starting to type or tabbing into that menu brings it up. However if you click on nothing it loses focus and once you stop moving you have to either start to type or double tab back to it.

**Comment by ninjanomnom** [comment]
Have to start to type to do what?

**Comment by Bawhoppen** [comment]
@Nabski89 You need to explain in-depth what the problem is. Don't assume we automatically have any clue what you're talking about.

**Comment by Nabski89** [comment]
I suspect I need to figure out how to make a gif for this to make sense.

Basically it randomly seems to switch between one mode and another for being able to controlwalk or not. The thing that decides if it switches is if the game things you're inputting text or not. If you're typing (indicated by the little blinking | bar), it works fine. The issue is that clicking else where on the screen makes that go away. So you'll be running fine, make an attack and miss keep running until you stop, then stand still and spin in retarded circles until you bring back up the blinking |.

**Comment by Nabski89** [comment]
I know this is more for the other PR that already got merged, but this fucks with pipecrawling really hard. You can diagonal move to exit the pipe at any time, and move between pipe systems (red and blue).

**Comment by ninjanomnom** [comment]
we know, there are plenty issues for it open

**Comment by Cyberboss** [comment]
Is this ready or..?

**Comment by nicbn** [comment]
Conflicts

**Pull request merged**